frb_codegen/src/parser/mod.rs

```diff
@@ -60,6 +61,8 @@ impl<'a> Parser<'a> {
     /// Attempts to parse the type from the return part of a function signature. There is a special
     /// case for top-level `Result` types.
     pub fn try_parse_fn_output_type(&mut self, ty: &syn::Type) -> Option<IrFuncOutput> {
+        let ty_alias = self.type_parser.get_alias_type(ty);
+        let ty = ty_alias.as_ref().unwrap_or(ty);
```

Comment:
seems this pattern (`get_alias(ty).unwrap_or(ty)`) is repeated a few times. shall we extract a function?

Reply:
I don't know what to do,
Also, because the `ty` here has to be ref, I am writinging it this way.

Reply:
One idea that might be possible,
Implement a function using a lifecycle, but I don't want to do that.

Reply:
Maybe use Cow?

e.g.

```
fn resolve_alias(ty: &syn::Type) -> Cow<syn::Type> {
  self.type_parser.get_alias_type(ty).map(Cow::Owned).unwrap_or(ty)
}
```

Reply:
I don't understand Cow, I just know it's clone-on-write. But there's no write here.
Is the following code OK?
@fzyzcjy 
```
pub fn resolve_alias(&self, ty: &syn::Type) -> Type {
        self.get_alias_type(ty).unwrap_or(ty).clone()
    }
    pub fn get_alias_type(&self, ty: &syn::Type) -> Option<&Type> {
        if let Type::Path(TypePath { qself: _, path }) = ty {
            if let Some(PathSegment {
                ident,
                arguments: _,
            }) = path.segments.first()
            {
                let key = &ident.to_string();
                if self.src_types.contains_key(key) {
                    return self.src_types.get(key);
                }
            }
        }
        None
    }
    pub fn parse_type(&mut self, ty: &syn::Type) -> IrType {
        let ty = &self.resolve_alias(ty);  // <<<<<<<
```
I wish I had more suggestions because this is a little difficult.

Reply:
Well maybe OK, given that Type is not very heavy. But cow is better.

As for what is cow: https://stackoverflow.com/questions/36706429/is-it-possible-to-return-either-a-borrowed-or-owned-type-in-rust

Reply:
I solved the problem, although I had the following problems before.
Thank you for your guidance, as well as my occasional successful fix
waiting for ci success
```rs
pub fn resolve_alias<'b: 'a>(&self, ty: &'b Type) -> Cow<Type> {
        Cow::Borrowed(self.get_alias_type(ty).unwrap_or(ty))
        // self.get_alias_type(ty)
        //     .map(|ty| Cow::Borrowed(ty))
        //     .unwrap_or(Cow::Borrowed(ty))
    }
    pub fn get_alias_type(&self, ty: &syn::Type) -> Option<&Type> {
        if let Type::Path(TypePath { qself: _, path }) = ty {
            if let Some(PathSegment {
                ident,
                arguments: _,
            }) = path.segments.first()
            {
                let key = &ident.to_string();
                if self.src_types.contains_key(key) {
                    return self.src_types.get(key);
                }
            }
        }
        None
    }
    pub fn parse_type(&mut self, ty: &syn::Type) -> IrType {
        let ty = &self.resolve_alias(ty);
        let supported_type = SupportedInnerType::try_from_syn_type(ty)
            .unwrap_or_else(|| panic!("Unsupported type `{}`", type_to_string(ty)));

        self.convert_to_ir_type(supported_type)
            .unwrap_or_else(|| panic!("parse_type failed for ty={}", type_to_string(ty)))
    }
```
```rs
 let ty = &self.resolve_alias(ty);
    |                   ---------------------- immutable borrow occurs here
...
175 |         self.convert_to_ir_type(supported_type)
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ mutable borrow occurs here
176 |             .unwrap_or_else(|| panic!("parse_type failed for ty={}", type_to_string(ty)))
```

Reply:
You are welcome. Btw I realize maybe `Cow<Type>` can be simply sth like `&Type`. (Originally I thought you mean you needed both owned and borrowed). Anyway this is just uber-nit.

Reply:
@fzyzcjy
There is only one job that fails.
Can you make it run again?
Or what can I do?

Reply:
Given another ios job succeeds, it is just flaky, no worries